packages/babel-helper-member-expression-to-functions/src/index.js

```diff
@@ -29,6 +29,38 @@ class AssignmentMemoiser {
   }
 }
 
+/**
+ * Test if a NodePath will be cast to boolean when evaluated.
+ *
+ * @example
+ * // returns true
+ * const nodePathAQDotB = NodePath("if (a?.#b) {}").get("test"); // a?.#b
+ * willPathCastToBoolean(nodePathAQDotB)
+ * @example
+ * // returns false
+ * willPathCastToBoolean(NodePath("a?.#b"))
+ * @todo Respect transparent expression wrappers
+ * @see {@link packages/babel-plugin-proposal-optional-chaining/src/index.js}
+ * @param {NodePath} path
+ * @returns {boolean}
+ */
+
+function willPathCastToBoolean(path: NodePath): boolean {
+  const maybeWrapped = path;
+  const { node, parentPath } = maybeWrapped;
+  if (parentPath.isLogicalExpression()) {
+    const { operator } = parentPath.node;
+    if (operator === "&&" || operator === "||") {
+      return willPathCastToBoolean(parentPath);
+    }
+  }
```

Comment:
We can also add the `operator === "??" && path.key === "right"` case

Reply:
Will address it in another commit.